### ISSUE TITLE
Validate channel

### ISSUE THREAD

**Issue by ctron** [ctron/package-drone#20]
Check if a channel  ~~can be resolved~~ is valid.

The original idea was to simply check if a P2 repository in a channel resolves. However over time a lot more ideas of validation came up. Many easier to implement in a first step.

So the goal is now more to create a way to validate channels in various ways, using aspects.

Of course the original use case of have a b3aggr like tool to validate the repository as well.

**Comment by ctron** [comment]
So this whole things gets a little clearer now. Package Drone needs a way to validate various aspects of its artifacts. I would say that the validation must take place on artifact and channel level. And can return at least "warning" and "error".

The question is what the result of this is. I could imagine a configurable policy per channel: show and fail. Where fail breaks the current operation (like upload of add channel aspect) and reverts to the last state.

**Comment by maggu2810** [comment]
Some ideas:
* All stuff that is triggered by the user and run in such a condition could result in some forced interaction (the user have to do something).
* If it is possible that error exists (the p2 aspect was added later) there could be a page such inform the user and he could choose some solutions (if possible).
* The p2 repo aspect should not fail at all, it should skip the bundles that generates such errors.

**Comment by ctron** [comment]
So I started playing around a little bit with validation. So for the moment I came up with the idea that there can be multiple validation results/messages with a severity of (INFO, WARNING, ERROR) for each channel, which can optionally reference to a list of artifacts (in the same channel).

This would allow for finding the bundle with a comma and marking it as "invalid", providing a message in addition. Since the aspect now has a way to hand over a state, it will also just ignore the bundle further on and not extract OSGi information.

Also in the case of the same channel having multiple artifacts with the same ID but different MD5 checksums, this should be a possible solution to report this, including having a list of the relevant artifacts.

For the moment there are dedicated access methods to get the validation messages from a channel. But I think there should be some sort of aggregated status which is pre-calculated and can easily be fetched with each lookup of a channel or aspect. Just in order to color the table entry of the channel/artifact in some color.

So this would make the user aware of the issue at least to some degree and provide detailed information of what went wrong.

I still think that there should be a more intrusive way to notify the user. Like @maggu2810 suggested.
